### PR TITLE
feat(jest) : setup jest

### PR DESCRIPTION
**Pull request by atticusofsparta**
Jest is now configured - properly - to work with vite, would like to merge setup-jest with vite branch before merging vite branch with develop.

### CODE REVIEW

**Comment by dtfiedler** [comment]
let's move this to PR against `develop` branch

File: README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Testing required a bit of configuration to use Jest with vite. We are using:
+
+- jest \ runs our tests
+- @testing-library/react \ renders components in our test env
+- @testing-library/jest-dom \ asserts components are in the dom and contain data
+- @testing-library/user-event \ programmatically interacts with components
+- @babel/preset-react >
+- @babel/preset-typescript >
+- @babel/preset-env \ these babel libs allow use to use JSX, TS, and ES6 modules in tests 
+- identity-obj-proxy \ helps with css modules so we can see original class names
```
Comment (dtfiedler) [inline README.md:34]: nit: not really necessary - you can obtain this from the `package.json` and the README should be focused on how to use, not necessarily dependencies we port in

File: README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+### Configuration
+
+the important config files are `babel.config.cjs` (cjs is CommonJS module extension) and the "jest" key in package.json.
+
+
+## File Structure
```
Comment (dtfiedler) [inline README.md:41]:
```suggestion
## Component Contribution
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+- When adding a translation, it goes in the assets/translations folder, named as <native-languages-name>.json. Translated text in the json should
+
+## Dependencies
+
+//todo
+
+## Translations
+
+//in progress
+we could use react-i18next lib for translations. It stores translations in JSON files and applies them when the locality is changed.
```
Comment (dtfiedler) [inline README.md:57]: nit: remove these for now. The README should ideally only have complete information

File: README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+- @testing-library/user-event \ programmatically interacts with components
+- @babel/preset-react >
+- @babel/preset-typescript >
+- @babel/preset-env \ these babel libs allow use to use JSX, TS, and ES6 modules in tests 
+- identity-obj-proxy \ helps with css modules so we can see original class names
+
+### Configuration
+
+the important config files are `babel.config.cjs` (cjs is CommonJS module extension) and the "jest" key in package.json.
+
```
Comment (dtfiedler) [inline README.md:39]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# arns-react
+A React App for the Arweave Name System Registry, which lets users search for and purchase Names.
+
+## Getting Started
+
+### Running the app locally
```
Comment (dtfiedler) [inline README.md:6]:
```suggestion
### Local Setup
```

File: package.json [hunk 6]
```diff
@@ -6,17 +6,45 @@
...
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^14.4.3",
     "@types/react": "^18.0.17",
     "@types/react-dom": "^18.0.6",
     "@vitejs/plugin-react": "^2.1.0",
+    "babel-jest": "^29.2.2",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^29.2.2",
+    "jest-environment-jsdom": "^29.2.2",
```
Comment (dtfiedler) [inline package.json:32]: let's also add `prettier` in this PR, we already have the configs in here

File: package.json [hunk 7]
```diff
@@ -6,17 +6,45 @@
   "scripts": {
     "dev": "vite",
     "build": "tsc && vite build",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "test": "jest"
```
Comment (dtfiedler) [inline package.json:10]: once prettier is added let's add one more script:

```
"format": "prettier --write .",
```

File: src/App.test.tsx [hunk 8]
```diff
@@ -0,0 +1,5 @@
+test("jest is working", ()=>{
+  expect(true).toBe(true);
+})
+
+export {}
```
Comment (dtfiedler) [inline src/App.test.tsx:5]: nit: no need to export on tests

File: coverage/lcov.info [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+TN:
+SF:src/main.tsx
+FNF:0
+FNH:0
+DA:6,0
+LF:1
+LH:0
+BRF:0
+BRH:0
+end_of_record
```
Comment (dtfiedler) [inline coverage/lcov.info:28]: let's a `coverage/` to .gitignore file

File: src/App.test.tsx (hunk 8 above)
Comment (atticusofsparta) [reply src/App.test.tsx:5]: You are correct - no need to export. Added this to clean up a vscode error in my syntax highlighting, still not quite sure how to fix that. But it works regardless.

File: README.md (hunk 1 above)
Comment (atticusofsparta) [reply README.md]: fixed in commit c09a9a10ff8bee85bb707e9b9796cc3f134a0a88

File: README.md (hunk 2 above)
Comment (atticusofsparta) [reply README.md]: fixed in commit e4eff7bd90681427abf826291072d15215748323

File: README.md (hunk 3 above)
Comment (atticusofsparta) [reply README.md]: fixed in commit c09a9a10ff8bee85bb707e9b9796cc3f134a0a88

File: README.md (hunk 5 above)
Comment (atticusofsparta) [reply README.md]: fixed in commit e4eff7bd90681427abf826291072d15215748323

File: package.json (hunk 7 above)
Comment (atticusofsparta) [reply package.json:11]: fixed in commit 41f3fe9644339bc8d444d2dff689b46fd1d9e35d

File: src/App.test.tsx (hunk 8 above)
Comment (atticusofsparta) [reply src/App.test.tsx]: fixed in commit 830e99a70897664a3d923b2d0f42c8e6053234cc

File: coverage/lcov.info (hunk 9 above)
Comment (atticusofsparta) [reply coverage/lcov.info]: removed coverage folder in commit d7b36a5fe217c1452e474749ffc9399785fdc224
added coverage to .gitignore in dfb740eb3fe353cca597f84748fe7c6f14098104

File: README.md (hunk 4 above)
Comment (atticusofsparta) [reply README.md]: fixed in commit 451e11ed3ec805a25e2ff5358913161595bcaf7b

**Review by dtfiedler — APPROVED** [review]
LG - nice work!

**Pull request merged**